src/main/java/com/cloudbees/jenkins/plugins/bitbucket/api/BitbucketApiFactory.java

```diff
@@ -44,20 +43,20 @@ protected abstract BitbucketApi create(@Nullable String serverUrl,
      * repository.
      *
      * @param serverUrl   the server URL.
-     * @param credentials the (optional) credentials.
+     * @param authenticator the (optional) authenticator.
      * @param owner       the owner name.
      * @param repository  the (optional) repository name.
      * @return the {@link BitbucketApi}.
      * @throws IllegalArgumentException if the supplied URL is not supported.
      */
     @NonNull
     public static BitbucketApi newInstance(@Nullable String serverUrl,
-                                           @Nullable StandardUsernamePasswordCredentials credentials,
+                                           @Nullable BitbucketAuthenticator authenticator,
```

Comment:
Changed public method signature. Should be fixable by having a method with the old signature that calls this with an authenticator instance created from the passed credentials.

Reply:
Someone might have written an extension, please consider adding the ability to accept StandardUsernamePasswordCredentials and convert them to BitbucketAuthenticator

Reply:
nvm already done